stock_dynamic_routing/models/stock_routing.py

```diff
@@ -146,6 +148,25 @@ def _compute_routing_message(self):
     def _default_sequence(self):
         return _default_sequence(self)
 
+    def _get_find_rule_for_location_domain(
+        self, pull_location_tree, push_location_tree, picking_type
+    ):
+        domain = expression.OR(
+            [
+                [
+                    ("routing_location_id", "in", pull_location_tree.ids),
+                    ("routing_picking_type_id", "=", picking_type.id),
+                    ("method", "=", "pull"),
+                ],
+                [
+                    ("routing_location_id", "in", push_location_tree.ids),
+                    ("routing_picking_type_id", "=", picking_type.id),
+                    ("method", "=", "push"),
+                ],
+            ]
+        )
+        return expression.AND([[("routing_id.active", "=", True)], domain])
```

Comment:
I think you then have to add auto_join=True on routing_id m2o definition

Reply:
We don't have to:
Without auto_join it creates a sql with `where routing_id in (select id from stock_routing where active=true)`
For example here a full query:

`SELECT "stock_routing_rule".id FROM "stock_routing_rule" WHERE (("stock_routing_rule"."routing_id" in (SELECT "stock_routing".id FROM "stock_routing" WHERE ("stock_routing"."active" = true))) AND (((("stock_routing_rule"."routing_location_id" in (5,2)) AND ("stock_routing_rule"."routing_picking_type_id" = 37)) AND ("stock_routing_rule"."method" = 'pull')) OR ((("stock_routing_rule"."routing_location_id" in (55,54,49,48,1)) AND ("stock_routing_rule"."routing_picking_type_id" = 37)) AND ("stock_routing_rule"."method" = 'push')))) ORDER BY  "stock_routing_rule"."sequence" ,"stock_routing_rule"."id"`

and with the auto_join it creates a sql with `left join stock_routing as sr .... where sr.active=true`
For example here also a full query
`SELECT "stock_routing_rule".id FROM "stock_routing_rule" LEFT JOIN "stock_routing" AS "stock_routing_rule__routing_id" ON ("stock_routing_rule"."routing_id" = "stock_routing_rule__routing_id"."id") WHERE (("stock_routing_rule__routing_id"."active" = true) AND (((("stock_routing_rule"."routing_location_id" in (63,58,57,1)) AND ("stock_routing_rule"."routing_picking_type_id" = 45)) AND ("stock_routing_rule"."method" = 'pull')) OR ((("stock_routing_rule"."routing_location_id" in (4,2)) AND ("stock_routing_rule"."routing_picking_type_id" = 45)) AND ("stock_routing_rule"."method" = 'push')))) ORDER BY  "stock_routing_rule"."sequence" ,"stock_routing_rule"."id"`